torch/onnx/_internal/fx/passes/decomp.py

```diff
@@ -69,14 +98,18 @@ def graph_with_interpreter(*args):
 
         fx_mode = "symbolic" if self.enable_dynamic_axes else "fake"
 
+        fn = wrapped_fn(
+            graph_with_interpreter, fx_mode, allow_non_fake_inputs=True, args=args
+        )
+
         # Apply decomposition table to the input graph.
         # Make sure the feed-in "module" is stateless.
         decomposed_module = proxy_tensor.make_fx(
-            graph_with_interpreter,
+            fn,
             decomposition_table=self.decomposition_table,
-            tracing_mode=fx_mode,
+            tracing_mode="real",
```

Comment:
`real`? I guess we always want to run decomposition with `fake` or `symbolic`. Maybe you can add another flag `reuse_existing_fake_or_create_fake`.?

Reply:
That was the flag I tried to add which people didn't want me to lol. There may be changes in the future to the api after this pr.

Reply:
Humm, I like that flag. If we change the value to `real`, exporter will be way slower.

Reply:
we're not actually doing real, we're still wrapping the inputs to fake. We're avoiding calling `real` so we dont do a new FakeTensorMode on top of the mode from the inputs.

Reply:
What if the input must be a FakeTensor?

In the error above, the whole model is initialized under FakeTensorMode, so all its parameters/buffers are FakeTensor's already. I guess that's why you see that assert. I am not sure how to deal with it without running the code. Building PyTorch on my side will take a while.

Reply:
need any pointers ?